framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -526,9 +601,11 @@ public JavaExpression visit(NameExpr expr, JavaExpressionContext context) {
          * to using that field at the given {@code context}. If {@code identifier} is not a field
          * name, this method returns null.
          *
-         * @param receiverExpr the receiver of the field; the expression used to access the field
+         * @param receiverExpr the receiver of the field access; the expression used to access the
+         *     field
          * @param identifier possibly a field name
-         * @return a field access, or null if {@code identifier} is not a field
+         * @return a field access, or null if {@code identifier} is not a field that can be accessed
+         *     via {@code receiverExpr}
          */
         protected @Nullable FieldAccess getIdentifierAsField(
                 JavaExpression receiverExpr, String identifier) {
```

Comment:
In this code below:
```java
            if (identifier.equals("length") && receiverType.getKind() == TypeKind.ARRAY) {
                fieldElem = resolver.findField(identifier, receiverType, pathToCompilationUnit);
            }
            if (fieldElem == null) {
```

1. Should there be an assertion that `fieldElem` is non-null after the assignment?
2. I think the `} if (fieldElem == null) {` can be changed to `} else {`.